---
title: "DMRscan Vignette"
author: "Laura Helmkamp and Robert Podolsky"
date: "`r Sys.Date()`"
output: BiocStyle::html_document
vignette: >
  %\VignetteIndexEntry{DMRscan Vignette}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---


1: Introduction
----------------------------------------------------------------------

DNA methylation is a crucial and commonly studied epigenetic mechanism for the regulation of gene expression.  Differentially methylated regions (DMRs) have been implicated in a number of diseases, as well as in a variety of developmental and aging processes.    

The **DMRscan** package allows users to accurately find DMRs using the scan statistic.  **DMRscan** constitues a convenient pipeline linking the sitewise likelihood-based differential methylation statistics calculated by the package **methylSig**,  the free scan statistic software **SaTScan** (via **RSaTScan**), and the plotting capabilities of **Gviz** [1,2,3,4].

This document is a step-by-step user guide for the DMRscan package.



2: Installation
----------------------------------------------------------------------


### 2a: Installing SaTScan ###

**DMRscan** requires the stand-alone software **SaTScan**.  This can be easily downloaded from http://www.satscan.org/download.html for Windows, Linux, or Mac OS X.   



### 2b: Installing DMRscan ###

Install **DMRscan** with the devtools R package:

```{r, message=F, warning=F,  eval=FALSE}
library(devtools) 
install_github('lhelmkamp/DMRscan')
```




3: Sample data and data class
----------------------------------------------------------------------

As sample data, we use a small portion of the sample data publicly available on the GEO database under accession number GSE61161.  Data was obtained from 39 patients with chronic myelomonocytic leukemia (CMML), and we are interested in locating regions which could predict response to a commonly used treatment.  Data for chromosome 18 is provided with the package. 

```{r, message=F, warning=F}
library(DMRscan) 

```

```{r, results = "hide", warning = FALSE, message=FALSE, echo = FALSE}
CMML_chr18

```


```{r, message=F, warning=F}
CMML_chr18[,1:5]

```


**DMRscan** requires that input data be formatted as methylSigData object, as is output from the *methylSigReadData()* function in the **methylSig** package.  If your data is not in this format, please see the Appendix.   


4: Finding DMRs with DMRscan
----------------------------------------------------------------------

With data in a methylSigData object, the function *DMRscan()* can be used to find DMRs.  *DMRscan()* performs the following steps:

1. Obtain sitewise likelihood ratio statistics
2. Normalize the sitewise statistics using quantiles
3. Write the files needed by SatScan to a local directory 
4. Run SatScan and view the results

We start with a simple call to *DMRscan()* for the chromosome 18 data:


```{r, eval=FALSE}
time0.a<-proc.time()
result.a<-DMRscan(CMML_chr18, xvalues = "Index")
time1.a<-proc.time()

```

Here, `xvalues = "Index"` indicates that in the scan statistic, the data points will be treated as equally spaced rather than using the position of the site on the chromosome.  We include timing code to allow the user to estimate computational time:

```{r, message=T, warning=F}
(time1.a-time0.a)/60
```

On our computer, where this test of chromosome 18 takes about 13 minutes, analyzing the entire genome-wide dataset for the CMML data takes about 16 hours.

The output `result.a` is a list, the first element of which contains the significant DMRs:

```{r, message=T, warning=F, results="hide"}
result.a$DMRs
```
```{r, echo=F}
result.a$DMRs[,c(1:7, 11)]
```

Note that the result is also automatically output to `result`; we output the result as a precaution due to the long computational time of *DMRscan()* on large datasets.  However, we recommend returning output to a more desciptive name as well, as `result` will be overwritten by subsequent calls to *DMRscan()*.


If multiple chromosomes are present in the input `meth` object, *DMRscan()* will automatically analyze each chromsome separately.  For human data, the user can also split each chromsome at the centromere by setting `splitcentromere = TRUE` and by specifying `build` from `"hg18"`, `"hg19"`, or `"hg38"`. This will speed up the analysis somewhat, and may be necesary for large datasets if system memory prevents the entire chromosome from being analyzed simultaneously.

```{r, eval=FALSE}
time0.b<-proc.time()
result.b<-DMRscan(CMML_chr18, xvalues = "Index", splitcentromere = TRUE, build = "hg18")
time1.b<-proc.time()

```

```{r, message=T, warning=F}
(time1.b-time0.b)/60
```

Comparing these new results to those obtained previously:

```{r, message=T, warning=F, results="hide"}
result.b$DMRs
```
```{r, echo=F}
result.b$DMRs[,c(1:7, 11)]
```

we note that the same number of regions was found, and the bounds of these regions are largely the same. However, the results have changed slightly. This is to expected, as SaTScan finds regions by minimizing the total variance of the data, and what we have specified as the "full" data has changed.


5: DMR visualization
----------------------------------------------------------------------

We can now visualize the significant regions with the function **plotSatScanresult**. The results will be output to a pdf in `mydir`, if this was specified, or in the current working directory. By default, all the significant results returned by **plotSatScanresult** will be plotted; the options `plottopn` and `plotpval` can be used to restrict plotting to the top few results by p-value or to results with a specified p-value more conservative than 0.05, respectively.  Plots can also be output directly from the call to **plotSatScanresult** by specifying `plotresult=TRUE`.  

Plotting uses Gviz to plot the DMR along with an ideogram showing the position of the DMR on the chromosome, the log-likelihood values of the data, and nearby genes, promoter regions, and CpG islands.  


```{r, eval=FALSE}
plotSatScanresult(result.b) 
```

For example:

```{r, include=FALSE}
result.plot<-result.b
result.plot$DMRs<-result.b$DMRs[7,]


#plotSatScanresult(result.plot) 

####################################### plot in window #################################
########## resume with the function
likelihoods<-result.b$likelihoods
build<-result.b$build
dir<-result.b$dir
DMRs<-result.plot$DMRs

library(rtracklayer)
library("GenomicFeatures")
library("Gviz")


downloadgenes<-makeTxDbFromUCSC( # this is a TxDb object, which is easy to plot with Gviz.
  genome=build,
  tablename="knownGene",
  transcript_ids=NULL,
  circ_seqs=DEFAULT_CIRC_SEQS,
  url="http://genome.ucsc.edu/cgi-bin/",
  goldenPath_url="http://hgdownload.cse.ucsc.edu/goldenPath",
  miRBaseBuild=NA
)


######## promoters from genes
my.promoters<-trim(suppressWarnings(promoters(downloadgenes, upstream=2000, downstream=200)), use.names=TRUE)

DMRs$chrnum<-as.numeric(gsub("chr", "", DMRs$chr))
RSatScanDMRs.order<-with(DMRs, DMRs[order(chrnum, pos.start),])



mychrindlist<-rownames(table(RSatScanDMRs.order$chr)) # this is not in order (1, 10, 11,...) instead of (1,2,..)
mychrlistorder<-intersect(c(paste("chr",1:22, sep=""),"chrX", "chrY"), mychrindlist )


DMRcount<-0
for (i in 1:length(mychrlistorder)){
  mychr<-mychrlistorder[i]
  chrDMRs<-which(RSatScanDMRs.order$chr==mychr)
  
  
  ##########################    + ideogram     ##########################  
  
  itrack <- IdeogramTrack(genome = build, chromosome = mychr) # this makes the ideogram track, given build and chromosome
  
  ##########################    + genes     ##########################  
  
  grtrack <- GeneRegionTrack(downloadgenes, genome = build,
                             name = "Genes", showId=FALSE, #geneSymbols = TRUE, 
                             chromosome=mychr
                             #from=region$start-5000, to=region$end+5000,
                             #transcriptAnnotation = "transcript" 
                             #,collapseTranscripts="meta"
  )
  
  #TranscriptDB does not have symbols?
  library("org.Hs.eg.db")
  symbols <- suppressWarnings(unlist(mapIds(org.Hs.eg.db, gene(grtrack), "SYMBOL", "ENTREZID",multiVals = "first")))
  
  
  
  
  a<-symbol(grtrack)
  b<-gene(grtrack)
  c<-symbols[gene(grtrack)]
  
  symbol(grtrack) <- symbols[gene(grtrack)]
  symbol(grtrack)[is.na(symbol(grtrack))]<-a[is.na(symbol(grtrack))]
  gene(grtrack)<- symbol(grtrack)  # this is new!
  ##########################    + promoters     ##########################  
  
  my.promoters.chr<-my.promoters[seqnames(my.promoters) == mychr  ]
  ptrack <- AnnotationTrack(my.promoters.chr, name = "Promoters")
  
  
  
  ##########################    + CpG islands     ##########################  
  
  CpGtrack=UcscTrack(track="CpG Islands", genome=build, chromosome=mychr, 
                     start="chromStart", end="chromEnd", name="CpG Islands", fill="#ADE6BA")
  
  
  
  for (DMRind in chrDMRs){  #
    RSatScanDMR<-RSatScanDMRs.order[DMRind,]
    DMRcount<-DMRcount+1
    cat("\n\n\n\n\n\nDMR", DMRcount, "of", dim(RSatScanDMRs.order)[1], "...\n\n\n\n\n\n")
    
    
    ##########################    + region     ##########################  
    
    region<-data.frame(RSatScanDMR["pos.start"],RSatScanDMR["pos.stop"], mychr)
    plotborder<-round((RSatScanDMR["pos.stop"]-RSatScanDMR["pos.start"])*1.0)
    plotstart<-max((RSatScanDMR["pos.start"]-plotborder), 0)
    plotstop<-min((RSatScanDMR["pos.stop"]+plotborder), end(tail(itrack@range@ranges, 1))  )
    
    
    names(region)<-c("start", "end", "chromosome")
    atrack <- AnnotationTrack(region, name = "DMR",  fill="#9966ff")
    
    
    ##########################    + likelihoods     ##########################  
    # this one is if we only want logliks in the region
    #mylogliks<-likelihoods[which(as.character(likelihoods$chr)==as.character(region$chromosome) & likelihoods$pos>=region$start  & likelihoods$pos<=region$end),]
    
    # this is if we want all logliks in the plotting area.
    mylogliks<-likelihoods[which(as.character(likelihoods$chr)==as.character(region$chromosome) & likelihoods$pos>=plotstart  & likelihoods$pos<=plotstop),]
    
    dtrack <- DataTrack(data = mylogliks$loglik, start = mylogliks$pos, end = mylogliks$pos, 
                        chromosome = as.character(region$chromosome), genome = build,
                        name = "LogLik", fill="#808080")
    
    
    ##########################    + gene names     ##########################  
    #grtrack@range@ranges #72163
    #symbol(grtrack)
    mygeneinds<-which(grtrack@range@ranges@start<=plotstop & (grtrack@range@ranges@start + grtrack@range@ranges@width -1 )>=plotstart)
    mygenes<-symbol(grtrack)[mygeneinds]
    namestarts<-pmax(plotstart+((plotstop-plotstart)*.12), (grtrack@range@ranges@start[mygeneinds]))
    
    mygenes.unique<-mygenes[!duplicated(mygenes)]
    namestarts.unique<-namestarts[!duplicated(mygenes)]
    
    ucgeneind<-sum(is.na(gene(grtrack)[mygeneinds]))
    
    if (length(namestarts)>0){
      nameTrack <- AnnotationTrack(start = namestarts.unique,
                                   width = ((plotstop-plotstart)*.1), chromosome = as.character(region$chromosome), group = mygenes.unique,
                                   genome = build, name = "Gene Names", showId = TRUE, 
                                   col=NULL, 
                                   fill="white")
    } else {nameTrack<- AnnotationTrack(start = plotstart,
                                        width = ((plotstop-plotstart)*.1), chromosome = as.character(region$chromosome), id = "empty",
                                        genome = build, name = "Gene Names", showFeatureId = FALSE, 
                                        col=NULL, 
                                        fill="white") 
    ucgeneind<-1
    }
  }}


```

```{r, echo=F, message=F,  fig.width=7, fig.height=9}


##########################    plot all     ##########################  
#failtracks<-list(itrack,  atrack, dtrack, grtrack, ptrack, CpGtrack )
#succeedtracks<-list(itrack,  atrack, dtrack, grtrack, ptrack, CpGtrack )
if(ucgeneind==0){
  plotTracks(list(itrack,  atrack, dtrack, grtrack, nameTrack, ptrack, CpGtrack ), from=plotstart, to=plotstop, geneSymbols = FALSE,
             fontsize=12, type = "histogram", 
             collapseTranscripts="meta" , 
             main=paste(RSatScanDMR$chr, ":",RSatScanDMR$pos.start, "-",RSatScanDMR$pos.stop ), 
             cex.main=0.75
  )
}




```



References
----------------------------------------------------------------------
1. Park, Y., Figueroa, M. E., Rozek, L. S. & Sartor, M. A. MethylSig: a whole genome DNA methylation analysis pipeline. Bioinformatics 30, 2414-22 (2014).

2. Kulldorff, M. SaTScan: Software for the spatial and space-time scan statistics. http://www.satscan.org/

3. Ken Kleinman (2015). rsatscan: Tools, Classes, and Methods for Interfacing with SaTScan Stand-Alone Software. R package version 0.3.9200.
   http://CRAN.R-project.org/package=rsatscan

4. Florian Hahne, Steffen Durinck, Robert Ivanek, Arne Mueller, Steve Lianoglou, Ge Tan and Lance Parsons (). Gviz: Plotting data and annotation information along genomic
   coordinates. R package version 1.12.1.




Appendix: Creating a methylSigData object
----------------------------------------------------------------------

A methylSigData object can be created from CpG methylation score files using the *methylSigReadData()* function in the **methylSig** package.  For a full description of the package and function options, see http://sartorlab.ccmb.med.umich.edu/software.  We will include a brief example here for simplicity.

In this example, we have previously downloaded all of the files from GSE into the working directory.

First we check the format of one of the text files.  If your data is not currently in subject-wise text files, data can be rearranged and output to text files that have the correct format.  From the MethylSig vignette:

>The CpG methylation score file must contain at least seven columns. Among these, second to seventh
>column must be, in order, chromosome, base, strand, coverage, percentage of Cs and percentage of Ts.
>Column names are not important. Strand format is F/R or +/-, where F/+ represents forward and R/-
>represents reverse strand.


```{r, eval=FALSE, include=FALSE}
setwd("H:/Methylation/Methods_comparison_GSE61161/data/")
```

```{r, eval=FALSE, include=TRUE}
fileList = dir()
subj1data_chr18<-read.table(fileList[1], header=TRUE)
```
```{r, eval=FALSE, include=FALSE}
subj1data_chr18<-testfile[which(testfile$chr=="chr18"),]
```

```{r, eval=TRUE, include=TRUE}
head(subj1data_chr18)
```

Now, read in the files with *methylSigReadData()*
```{r, eval=FALSE}
sample.id = substr(fileList, 1, 10)
responder<-c(0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 0, 0, 1, 0, 0, 1, 0, 0, 0, 0, 0, 1, 1, 1, 1, 0, 0, 0, 1, 0, 1, 0, 1, 1, 1, 1, 1, 1)

#### Read Data ####
memory.limit(25000)
meth <- methylSigReadData(fileList, sample.ids = sample.id, assembly = "hg18", treatment = responder, context = "CpG", destranded=TRUE)

```


You are now ready to run DMRscan!
